### DISCUSSION TITLE
Is the useAsyncGql `limit` option supposed to work out of the box?

### DISCUSSION THREAD

**Discussion by TheDutchCoder** [Diizzayy/nuxt-graphql-client#190] [Q&A] [answered]
In the docs there's an example with a `limit` option: `const { data } = await useAsyncGql('launches', { limit: 5 });`
However when trying to use that there are 2 issues:

1. A TypseScript error: `Type 'number' is not assignable to type 'never'. ts(2322)`
2. It doesn't actually do anything

The alternative syntax throws even more TS errors (no overload matches this call):
```ts
const { data } = await useAsyncGql({
  operation: 'getCollections',
  variables: { limit: 2 }
})
```

**Comment by Diizzayy** [comment, marked as answer]
@TheDutchCoder Indeed, it does in fact work out the box, however though it expects for there a be a query defined as seen below:

```gql
query launches($limit: Int = 10) {
  launches(limit: $limit, find: { mission_name: "Starlink" }) {
    id
    mission_name
  }
}
```

**Reply by TheDutchCoder** [reply]
Oh god, it's just a variable haha.
For some reason I thought it was a way to limit graph results and got confused.

Thanks as always!
